iceoryx2-pal/posix/src/c/socket_macros.c

```diff
@@ -1,11 +1,24 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
 #ifdef _WIN64
+// clang-format off
+// the include order is important, since some headers are defining macros that
+// are used in the next header
 #include <WinSock2.h>
 #include <Windows.h>
 #include <MSWSock.h>
 #include <io.h>
```

Comment:
Oh boy. I'm already looking forward to implement MinGW support for iceoryx2 :sweat_smile:

Reply:
Hehe those things are a lot of fun, and they remember me of my old university job where bash script A declared a variable, that was modified in bash script B and C and D used it and when for whatever reason the scripts were called in different order weird things happened.